AEPServices/Sources/utility/ThreadSafeArray.swift

```diff
@@ -36,6 +36,17 @@ public final class ThreadSafeArray<T> {
         }
     }
 
+    /// Removes and returns the first element of the thread safe array.
+    /// Returns nil if the array is empty
+    public func removeFirst() -> T? {
+        queue.sync {
+            if array.isEmpty {
+                return nil
+            }
+            return self.array.removeFirst()
```

Comment:
what's the behavior of this method if `self.array` is empty?

Reply:
It crashes :(

Reply:
And I realized better to handle the crash and return optional